rollup.config.js

```diff
@@ -17,46 +17,77 @@ function toCamelCase(name) {
   return name.replace(/-(\w)/g, (match, letter) => letter.toUpperCase())
 }
 
-const base = {
-  input: path.join(rootPath, './src/index.js'),
-  plugins: [
-    nodeResolve(),
-    babel({
-      exclude: ['**/node_modules/**'],
-      babelrc: false,
-      presets: [['env', {modules: false}], 'stage-0', 'flow'],
-      plugins: ['external-helpers']
-    }),
-    commonjs({ignoreGlobal: true}),
-    globals(),
-    replace({
-      'process.env.NODE_ENV': JSON.stringify('development'),
-      'process.env.VERSION': JSON.stringify(pkg.version)
-    }),
-    sizeSnapshot({
-      matchSnapshot,
-      snapshotPath: './.size-snapshot.json'
-    })
-  ]
+const input = path.join(rootPath, './src/index.js')
+
+const name = toCamelCase(pkg.name)
+
+const globals = Object.keys(pkg.peerDependencies || {}).reduce(
+  (acc, key) => ({...acc, [key]: toCamelCase(key)}),
+  {}
+)
+
+const getBabelOptions = () => ({
+  exclude: '**/node_modules/**',
+  babelrc: false,
+  presets: [['env', {modules: false}], 'stage-0', 'flow'],
+  plugins: ['external-helpers']
+})
+
+const commonjsOptions = {
+  ignoreGlobal: true
+}
+
+const snapshotOptions = {
+  matchSnapshot,
+  snapshotPath: './.size-snapshot.json'
 }
 
 export default [
-  Object.assign({}, base, {
+  {
+    input,
     output: {
       file: `dist/${pkg.name}.js`,
       format: 'umd',
       sourcemap: true,
       exports: 'named',
-      name: toCamelCase(pkg.name)
-    }
-  }),
-  Object.assign({}, base, {
-    plugins: [].concat(base.plugins, uglify()),
+      name,
+      globals
+    },
+    external: Object.keys(globals),
+    plugins: [
```

Comment:
I rewrote last time this thing to have same plugins in one place, because there were already problems, since one needs to do everything twice with the plugins and can be easily forgotten

can we please have just one place for the plugins?

Reply:
In that refactoring you replaced NODE_ENV with `development` for both umd and umd.min bundles. The case is that plugins can be different. It's cleaner to keep them separated and reuse only options.

I'm gonna add cjs and esm bundles in the next PR and their configs will be also different.

Reply:
oh, i just forgot to make different replace plkugins for both …  we will still have many same plugins though

Reply:
And I wouldn't add more plugins. These lists won't change for a long time.

Reply:
I bet someone will break next time one of the environments again if we don't put the same plugins with setup in one place

Reply:
In the bright future we will remove `commonjs` and `node-globals` plugins